setup.py

```diff
@@ -0,0 +1,69 @@
+from setuptools import setup, Extension
+from Cython.Build import cythonize
+import os
+from Cython.Distutils import build_ext as cython_build_ext
+import multiprocessing
+
+os.environ["CFLAGS"] = "-s -flto"
+os.environ["LDFLAGS"] = "-flto"
+# Define the base directory
+base_dir = "pyparsing"
+
+# Find all .py files in the pyparsing package recursively
+py_files = []
+for root, dirs, files in os.walk(base_dir):
+    for file in files:
+        if file.endswith(".py"):
+            py_files.append(os.path.join(root, file))
+
+# Compiler directives
+base_compiler_directives = {
+    "language_level": 3,
```

Comment:
cython's defualt language_level is '3str'. Why not use the default?

Reply:
Would this also not require to comment out the unicode literals?
@Mrogovp could you please check?